src/main/java/org/javarosa/core/util/ArrayUtilities.java

```diff
@@ -33,6 +35,20 @@ public static boolean arraysEqual(byte[] array1, byte[] array2) {
     }
 
     /**
+     * Reverses an Array
+     */
+    public static <T> T[] reverse(Object[] myArray, Class<? extends T[]> tClass) {
```

Comment:
Another option is if you start with splitting your `input` String to the `verhoeffChecksum` function into an `List` instead of an array, then you can use `Collections.reverse()`